python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -123,45 +133,40 @@ def launch_run(self, context: LaunchRunContext) -> None:
         )
 
         arn = response["tasks"][0]["taskArn"]
-        self._instance.add_run_tags(run.run_id, self._run_tags(task_arn=arn))
-        self.ecs.tag_resource(resourceArn=arn, tags=self._ecs_tags(run.run_id))
+        self._set_run_tags(run.run_id, task_arn=arn)
+        self._set_ecs_tags(run.run_id, task_arn=arn)
         self._instance.report_engine_event(
             message=f"Launching run in task {arn} on cluster {metadata.cluster}",
             pipeline_run=run,
             cls=self.__class__,
         )
 
     def can_terminate(self, run_id):
-        tags = self._instance.get_run_by_id(run_id).tags
-        arn = tags.get("ecs/task_arn")
-        cluster = tags.get("ecs/cluster")
-
+        arn, cluster = self._get_run_tags(run_id)
         if arn and cluster:
-            status = (
-                self.ecs.describe_tasks(tasks=[arn], cluster=cluster)
-                .get("tasks", [{}])[0]
-                .get("lastStatus")
-            )
+            tasks = self.ecs.describe_tasks(tasks=[arn], cluster=cluster).get("tasks")
+            if not tasks:
+                return False
+            status = tasks[0].get("lastStatus")
             if status and status != "STOPPED":
                 return True
 
         return False
```

Comment:
very small nit: this conditional has the main of the logic of this function. would prefer that it is deindented.

```suggestion
    if not (arn and cluster):
        return False
    tasks = self.ecs.describe_tasks(tasks=[arn], cluster=cluster).get("tasks")
    if not tasks:
        return False
    status = tasks[0].get("lastStatus")
    if status and status != "STOPPED":
        return True
```